src/Storages/System/StorageSystemZooKeeper.cpp

```diff
@@ -176,18 +176,56 @@ static Paths extractPath(const ASTPtr & query, ContextPtr context)
     return extractPathImpl(*select.where(), res, context) ? res : Paths();
 }
 
+static void addRow(MutableColumns & res_columns, const String& node, const String& path, const Coordination::GetResponse& res)
+{
+    if (res.error == Coordination::Error::ZNONODE)
```

Comment:
From my perspective it is better to move this check into method caller, that way method `addRow` and change argument from `Coordination::GetResponse` to `Coordination::Stat`, that way method will be responsible only for inserting statistics into result columns.